pidtree_bcc/utils.py

```diff
@@ -138,3 +139,8 @@ def get_network_namespace(pid: int = None) -> int:
         return int(ns_link.strip()[5:-1])
     except Exception:
         return None
+
+
+def self_restart():
```

Comment:
Not technically a restart though - I think either we should rename this to self_terminate or something similar to avoid confusion, or alternatively change the implementation to stop all threads and then have the process replace itself with an exec.

Not a hard opinion though so feel free to ignore if there's no tidy solution.

Reply:
Well, I think it's actually much closer to a "restart" than a "terminate", since what it causes is for all processes/threads except one to stop and then restarts them. The execve path was an option, but this way is cleaner in my opinion, and surely faster.